code/modules/mob/living/brain/MMI.dm

```diff
@@ -127,25 +127,30 @@
 
 
 /obj/item/mmi/proc/transfer_identity(mob/living/L) //Same deal as the regular brain proc. Used for human-->robot people.
-	if(!brainmob)
-		brainmob = new(src)
-	brainmob.name = L.real_name
-	brainmob.real_name = L.real_name
-	if(L.has_dna())
-		var/mob/living/carbon/C = L
-		if(!brainmob.stored_dna)
-			brainmob.stored_dna = new /datum/dna/stored(brainmob)
-		C.dna.copy_dna(brainmob.stored_dna)
-	brainmob.container = src
-
 	if(ishuman(L))
 		var/mob/living/carbon/human/H = L
 		var/obj/item/organ/brain/newbrain = H.getorgan(/obj/item/organ/brain)
-		newbrain.forceMove(src)
-		brain = newbrain
-	else if(!brain)
+		if(newbrain)
+			. = TRUE
+			newbrain.Remove(H, TRUE) //this calls newbrain.transfer_identity()
```

Comment:
Why remove and not just transfer_identity?

Reply:
`h.Remove()` removes the brain from the human's list of organs so that it doesn't get qdeleted along with the mob